python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py

```diff
@@ -0,0 +1,464 @@
+        result_series = pd.concat([event_series, result_series])
+
+        # Display the resulting Series
+        return result_series
+
+
+def find_adjacent_true_regions(
+    series: pd.Series, min_adjacent: None | int = None
+) -> list[tuple[pd.DatetimeIndex, pd.DatetimeIndex]]:
+    """
+    Find regions in a boolean Series containing adjacent True values.
+
+    Parameters
+    ----------
+    series : pd.Series
+        The boolean Series to search for regions.
+
+    min_adjacent : int, optional
+        Minimum number of adjacent True values in a region.
+        Defaults to half size of the series.
+
+    Returns
+    -------
+    list[tuple[pd.DatetimeIndex, pd.DatetimeIndex]]
+        A list of tuples representing the start and end indices of regions
+        containing more than or equal to min_adjacent adjacent True values.
+    """
+    min_adjacent = min_adjacent if min_adjacent else 0.5 * series.size
+    regions = []
+    for key, group in series.groupby((series != series.shift()).cumsum()):
+        if key and len(group) >= min_adjacent:
+            region_indices = group.index
+            regions.append((region_indices.min(), region_indices.max()))
+    return regions
+
+
+def get_tma_slew_event(day_obs: int, seq_number: int) -> TMAEvent:
+    """
+    Retrieve Telescope Mount Assembly (TMA) slew events within a specified time
+    range.
+
+    Parameters
+    ----------
+    dayObs : int
+        Observation day in the YYYYMMDD format.
+    seqNum : int
+        Sequence number associated with the slew event.
+
+    Returns
+    -------
+    lsst.summit.utils.tmaUtils.TMAEvent
+        A TMA slew events that occurred within the specified time range.
+
+    Notes
+    -----
+    This function retrieves TMA slew events occurring between the specified
+    start and end times. It uses the TMAEventMaker class to obtain events for
+    the specified day of observation (dayObs). The events are filtered to
+    include only those that start after 1 second before the specified start
+    time and end before 1 second after the specified end time.
+
+    Returs
+    ------
+    lsst.summit.utils.tmaUtils.TMAEvent
+        A TMA slew event that occurred within the specified time range.
+
+    Raises
+    ------
+    ValueError
+        If no events are found for the specified time range.
+    ValueError
+        If more than one event is found for the specified time range.
+    """
+    logger.info(f"Query events in {day_obs}")
+    events = event_maker.getEvents(day_obs)
+
+    logger.info(f"Found {len(events)} events.")
+    single_event = [e for e in events if e.seqNum == seq_number]
+
+    logger.info(f"Found {len(single_event)} matching event(s).")
+    if len(single_event) > 1:
+        raise ValueError(
+            f"Expected a single event for {day_obs}. "
+            f"Found {len(single_event)} events."
+        )
+
+    if len(single_event) == 0:
+        raise ValueError(f"Could not find any events for {day_obs}. ")
```

Comment:
This error message isn't accurate - if `len(events) == 0` then it could not find any events, but as written if you ask for event one million on a given day, you'll get this error even if hundreds are found.

Reply:
I created a new check a few lines above and changed the message in this line. Hope it is clearer now.